### PR TITLE
Build docker image with gradle

### PR DESCRIPTION
**Pull request by MrCreosote**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,15 +2,27 @@ FROM eclipse-temurin:11-jdk as build
 
 WORKDIR /tmp
```
Comment (Xiangs18) [inline Dockerfile:3]: this line could be removed as well?

Comment (MrCreosote) [reply Dockerfile:3]: No, the ./gradlew war command depends on it

File: Dockerfile [hunk 2]
```diff
@@ -2,15 +2,27 @@ FROM eclipse-temurin:11-jdk as build
...
+RUN ./gradlew dependencies
+
+# Now build the code
+COPY workspace.spec /tmp/workspace/workspace.spec
+COPY deployment/ /tmp/workspace/deployment/
+COPY docshtml /tmp/workspace/docshtml/
+COPY docsource /tmp/workspace/docsource/
+COPY lib /tmp/workspace/lib/
+COPY src /tmp/workspace/src/
+COPY war /tmp/workspace/war/
```
Comment (Xiangs18) [inline Dockerfile:23]: Already set `WORKDIR /tmp/workspace`, maybe we can simply `/tmp/workspace` as `.` when doing copy

File: Dockerfile (hunk 1 above)
Comment (MrCreosote) [reply Dockerfile]: Fixed

File: Dockerfile (hunk 2 above)
Comment (MrCreosote) [reply Dockerfile:22]: I would rather keep paths absolute and explicit when possible. `gradlew` needs to run in the project directory AIUI

**Comment by MrCreosote** [comment]
Codecov report is here: https://github.com/kbase/workspace_deluxe/pull/729#issuecomment-2048234433

**Comment by MrCreosote** [comment]
It seems like it's no longer possible to run codecov from a fork

**Review by Xiangs18 — APPROVED** [review]
LGTM!

**Review by briehl — APPROVED** [review]

**Pull request merged**